feature-libs/cart/wish-list/core/store/actions/wish-list.action.ts

```diff
@@ -41,57 +39,29 @@ export class CreateWishListFail extends StateUtils.EntityFailAction {
 interface LoadWishListPayload {
   userId: string;
   /**
-   * Used to compute wishlist cart name and find it in list of all carts.
+   * This is tempCartId which is exact the wishlist name computed from customerId
    */
-  customerId: string;
-  /**
-   * When we try load wishlist for the first time we don't know cart id.
-   * Instead we create temporary cart with id equal to wishlist name to keep track of loading/error state.
-   */
-  tempCartId: string;
+  cartId: string;
 }
 
+/**
+ * When we try load wishlist for the first time we don't know cart id.
+ * Instead we create temporary cartId from wishlist name to keep track of loading/error state.
+ */
 export class LoadWishList extends StateUtils.EntityLoadAction {
   readonly type = LOAD_WISH_LIST;
   constructor(public payload: LoadWishListPayload) {
-    super(MULTI_CART_DATA, payload.tempCartId);
+    super(MULTI_CART_DATA, payload.cartId);
   }
 }
-
-interface LoadWishListSuccessPayload {
-  cart: Cart;
-  userId: string;
-  /**
-   * When LoadWishListSuccess action was dispatched as an completion to LoadWishList action
-   * we get temporary cartId that was used to keep track of loading state.
-   * In case of loading wish list with known cartId this property will be empty.
-   */
-  tempCartId?: string;
-  /**
-   * Used to compute wishlist cart name and find it in list of all carts.
-   * In case of loading wish list with known cartId this property will be empty.
-   */
-  customerId?: string;
-  /**
-   * Wish list cart id. Extracted from cart content (code property).
-   */
-  cartId: string;
-}
-
 export class LoadWishListSuccess extends StateUtils.EntitySuccessAction {
   readonly type = LOAD_WISH_LIST_SUCCESS;
-  constructor(public payload: LoadWishListSuccessPayload) {
+  constructor(public payload: { cart: Cart; cartId: string }) {
```

Comment:
Isn't the `cartId` in the `cart` already?

Reply:
cartId can be cart's code or cart's guid. 
```
new WishListActions.LoadWishListSuccess({
                  cart: wishList,
                  cartId: getCartIdByUserId(wishList, userId),
                })
```
I didn't change this codes, but anonymous user cannot have wishlist.